### PR TITLE
Add invoicing parameters to Subscription and Invoice

### PR DESCRIPTION
**Pull request by jacqueline-stripe**
This PR adds and tests new invoicing related parameters:
Subscription model: billing, days_until_due
Invoice model: billing, due_date, number

I don't believe any changes need to be made for what parameters are accepted in function calls.

### CODE REVIEW

**Comment by brandur-stripe** [comment]
@jacqueline-stripe Thanks! This looks generally good to me.

I notice that all these fields are still marked as `dashboard_only` and gated internally. I assume all of these are coming off?

> TODO: investigate tests are correct

Can you see if you can run them successfully and let me know?

**Comment by jacqueline-stripe** [comment]
@brandur-stripe the gates will come off when we launch (Launching the bindings and the feature is a bit of a chicken-and-egg problem). cc @stripe/api-libraries 

r? @pantera-stripe

**Comment by brandur-stripe** [comment]
> @brandur-stripe the gates will come off when we launch (Launching the bindings and the feature is a bit of a chicken-and-egg problem)

Cool! Sounds good.

File: src/main/java/com/stripe/model/Invoice.java [hunk 1]
```diff
@@ -25,12 +26,14 @@
 	Long date;
 	String description;
 	Discount discount;
+	Long dueDate;
```
Comment (pantera-stripe) [inline src/main/java/com/stripe/model/Invoice.java]: The API docs also list `days_until_due` as a create parameter. Is there a reason we're only exposing  the due date parameter?

File: src/test/java/com/stripe/functional/InvoiceTest.java [hunk 2]
```diff
@@ -256,4 +258,22 @@ public void testInvoiceMetadata() throws StripeException {
...
+        params.put("billing", "send_invoice");
+        params.put("due_date", dueDate);
+        Invoice invoice = Invoice.create(params);
+        assertEquals(invoice.getBilling(), "send_invoice");
+        assertEquals(invoice.getDueDate(), dueDate);
+
+        Map<String,Object> updateParams = new HashMap<String,Object>();
+        updateParams.put("paid", true);
+        Invoice updatedInvoice = invoice.update(updateParams);
+        assertEquals(updatedInvoice.getPaid(), true);
```
Comment (pantera-stripe) [inline src/test/java/com/stripe/functional/InvoiceTest.java]: You can just use `assertTrue(updatedInvoice.getPaid())` here.

File: src/test/java/com/stripe/functional/InvoiceTest.java [hunk 3]
```diff
@@ -256,4 +258,22 @@ public void testInvoiceMetadata() throws StripeException {
...
+    public void testSendInvoice() throws StripeException {
+        InvoiceItem invItem = InvoiceItem.create(getInvoiceItemParams());
+        Map<String,Object> params = new HashMap<String,Object>();
+        Long dueDate = (System.currentTimeMillis() / 1000) + 600;
+        params.put("customer", invItem.getCustomer());
+        params.put("billing", "send_invoice");
+        params.put("due_date", dueDate);
+        Invoice invoice = Invoice.create(params);
+        assertEquals(invoice.getBilling(), "send_invoice");
+        assertEquals(invoice.getDueDate(), dueDate);
```
Comment (pantera-stripe) [inline src/test/java/com/stripe/functional/InvoiceTest.java]: Assertion order reversed on both of these assertions. Should be `assertEquals(expect, actual)` http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertEquals(java.lang.Object, java.lang.Object)

File: src/test/java/com/stripe/functional/SubscriptionTest.java [hunk 4]
```diff
@@ -192,4 +192,26 @@ public void testSubscriptionMetadata() throws StripeException {
...
+        Map<String, Object> subscriptionParams = new HashMap<String, Object>();
+        subscriptionParams.put("plan", plan.getId());
+        subscriptionParams.put("billing", "send_invoice");
+        subscriptionParams.put("days_until_due", 30);
+        subscriptionParams.put("customer", customer.getId());
+        Subscription sub = Subscription.create(subscriptionParams);
+        assertEquals(sub.getPlan().getId(), plan.getId());
+        assertEquals(sub.getBilling(), "send_invoice");
+        assertEquals(sub.getDaysUntilDue(), (Integer) 30);
+
```
Comment (pantera-stripe) [inline src/test/java/com/stripe/functional/SubscriptionTest.java]: Assertion orders reversed again.

File: src/test/java/com/stripe/functional/SubscriptionTest.java [hunk 5]
```diff
@@ -192,4 +192,26 @@ public void testSubscriptionMetadata() throws StripeException {
...
+        Map<String, Object> updateParams = new HashMap<String, Object>();
+        updateParams.put("days_until_due", 10);
+        Subscription subUpdated = sub.update(updateParams);
+        assertEquals(subUpdated.getDaysUntilDue(), (Integer) 10);
```
Comment (pantera-stripe) [inline src/test/java/com/stripe/functional/SubscriptionTest.java]: Same.

**Comment by pantera-stripe** [comment]
ptal @jacqueline-stripe

File: src/main/java/com/stripe/model/Invoice.java (hunk 1 above)
Comment (jacqueline-stripe) [inline src/main/java/com/stripe/model/Invoice.java]: I believe the Java bindings only require you to add the fields on the model, not the parameters accepted to API calls. `days_until_due` is a parameter that can be passed on create, but is not a field on the invoice model.

**Comment by pantera-stripe** [comment]
I'm Pantera, and I approve this PR.

**Comment by jacqueline-stripe** [comment]
@brandur-stripe feel free to release as early as 2/22 (We will not be launching to public until March, but similar to Multiplan Subscriptions, I believe it should be ok to release the bindings for people to test out).

**Comment by brandur-stripe** [comment]
Released as 3.11.0.

**Pull request merged**